File: src/main/java/niklasu/speedtester/downloader/DownloaderModule.java
package niklasu.speedtester.downloader;

import java.net.MalformedURLException;
import java.net.URL;

import com.google.inject.AbstractModule;
import com.google.inject.Provides;
import niklasu.speedtester.config.ConfigProvider;
import niklasu.speedtester.influx.InfluxModule;
import niklasu.speedtester.ui.ConsoleResultPrinter;

public class DownloaderModule extends AbstractModule {
    @Provides
    private DownloadThread getThread(ConfigProvider configProvider, ConsoleResultPrinter consoleResultPrinter,
            InfluxModule.InfluxWriter influxConnector)
            throws MalformedURLException {
        return new DownloadThread(new URL(configProvider.url), configProvider.size, consoleResultPrinter,
                influxConnector);
    }

}

File: src/main/java/niklasu/speedtester/downloader/DownloaderModule.kt
package niklasu.speedtester.downloader

import com.google.inject.AbstractModule
import com.google.inject.Provides
import niklasu.speedtester.config.ConfigProvider
import niklasu.speedtester.measurements.Measurements
import niklasu.speedtester.ui.ConsoleResultPrinter
import java.net.URL


class DownloaderModule : AbstractModule() {
    @Provides
    private fun getThread(configProvider: ConfigProvider, consoleResultPrinter: ConsoleResultPrinter, measurements: Measurements): DownloadThread {
        return DownloadThread(URL(configProvider.url), configProvider.size, consoleResultPrinter, measurements)
    }

}